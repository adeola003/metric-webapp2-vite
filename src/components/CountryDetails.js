import React, { useEffect } from 'react';
import { useParams, Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { searchByCode } from '../redux/countries/countriesSlice';

const CountryDetails = () => {
  const { loading, countryResult } = useSelector((store) => store.countries);
  const dispatch = useDispatch();
  const { code } = useParams();

  useEffect(() => {
    if (code) {
      console.log(code);
      dispatch(searchByCode(code.toLowerCase()));
    }
  }, [dispatch, code]);

  if (loading || countryResult.length === 0) {
    return (<h2>Content is not fetched thus not available</h2>);
  }
  return (
    <section className='country-details'>
      <div>
        <h1>{countryResult[0].name.official}</h1>
        <img src={countryResult[0].flags.png} alt={`Flag of ${countryResult[0].name.common}`} />
        <p>
          Capital:&nbsp;
          {countryResult[0].capital}
        </p>
        <p>
          <stron>Capital Info:</stron>
          <span>
            {' '}
            Latitude_&nbsp;
            {countryResult[0].capitalInfo.latlng[0]}
          </span>
          ;
          <span>
            {' '}
            Longitude_&nbsp;
            {countryResult[0].capitalInfo.latlng[1]}
          </span>
        </p>
        <p>
          Continent:&nbsp;
          {countryResult[0].continents[0]}
        </p>
        <p>
          Currencies:&nbsp;
          {Object.values(countryResult[0].currencies).map((currency) => currency.name).join(', ')}
        </p>
        <p>
          Languages:&nbsp;
          {Object.values(countryResult[0].languages).join(', ')}
        </p>
        <p>
          Region:&nbsp;
          {countryResult[0].region}
        </p>
        <p>
          Population:&nbsp;
          {countryResult[0].population}
        </p>
        <p>
          Timezones:&nbsp;
          {countryResult[0].timezones.join(', ')}
        </p>
        <div className="borders">
          <p>Country borders:&nbsp;</p>
          {countryResult[0].borders ? (countryResult[0].borders.map((border) => (
            <Link className="border-name" to={`/${border}`} key={border}>
              <p>&nbsp;{border}&nbsp;</p>
            </Link>
          ))
          )
            : (<span>No borders</span>)}

        </div>
      </div>
    </section>
  );
};

export default CountryDetails;
