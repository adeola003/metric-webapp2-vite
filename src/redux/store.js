import { configureStore } from '@reduxjs/toolkit';
import { countriesReducer } from './countries/countriesSlice'; // Update the import statement

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
