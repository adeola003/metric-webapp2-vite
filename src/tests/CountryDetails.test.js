import React from "react";
import renderer from "react-test-renderer";
import store from "../redux/store";
import CountryDetails from "../components/CountryDetails";
import { Provider } from "react-redux";


describe('Test Rockets component', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <CountryDetails />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });