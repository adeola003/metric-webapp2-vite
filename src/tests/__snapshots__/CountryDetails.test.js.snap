// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rockets component renders correctly 1`] = `
<h2>
  Content is not fetched thus not available
</h2>
`;
